antsibull/docs_parsing/routing.py

```diff
@@ -0,0 +1,326 @@
+    """
+    Calculate source and dest FQCNs for plugins which have been renamed via symlink.
+
+    :collection_name: FQCN of the collection
+    :src_basename: Alias name for the plugin
+    :dst_basename: Canonical name of the plugin
+    :rel_path: Filesystem path to the directory that the plugin lives in relative to the plugin
+        type directory.
+    :returns: Two-tuple of fqcn of the alias name of the plugin and fqcn of the canonical
+        plugin name.
+    """
+    src_components = os.path.normpath(
+        os.path.join(rel_path, src_basename)).split(os.sep)
+    dest_components = os.path.normpath(
+        os.path.join(rel_path, dst_basename)).split(os.sep)
+
+    # Compose source FQCN
+    src_name = '.'.join(src_components)
+    src_fqcn = f'{collection_name}.{src_name}'
+
+    # Compose destination FQCN
+    dst_name = '.'.join(dest_components)
+    dst_fqcn = f'{collection_name}.{dst_name}'
+
+    return src_fqcn, dst_fqcn
+
+
+def find_symlink_redirects(collection_name: str,
+                           plugin_type: str,
+                           directory_path: str
+                           ) -> t.Dict[str, str]:
+    """
+    Finds plugin redirects that are defined by symlinks.
+
+    :collection_name: FQCN of the collection we're searching within
+    :plugin_type: Type of plugins that we're scanning
+    :directory_path: Full path to the directory we're scanning.
+    :returns: Dict mapping fqcn of the alias names to fqcn of the canonical plugin names.
+    """
+    plugin_type_routing = dict()
+    if os.path.isdir(directory_path):
+        for path, dummy, files in os.walk(directory_path):
+            rel_path = os.path.relpath(path, directory_path)
+            for filename in files:
+                src_basename, ext = os.path.splitext(filename)
+                if ext != '.py' and not (plugin_type == 'module' and ext == '.ps1'):
+                    continue
+
+                file_path = os.path.join(path, filename)
+                if not os.path.islink(file_path):
+                    continue
+
+                dst_basename = os.path.splitext(os.readlink(file_path))[0]
+
+                src_fqcn, dst_fqcn = calculate_plugin_fqcns(collection_name, src_basename,
+                                                            dst_basename, rel_path)
+
+                plugin_type_routing[src_fqcn] = dst_fqcn
+
+    return plugin_type_routing
+
+
+def process_dates(plugin_record: t.Dict[str, t.Any]) -> t.Dict[str, t.Any]:
+    for tlkey in ('tombstone', 'deprecation'):
+        if tlkey in plugin_record and 'removal_date' in plugin_record[tlkey]:
+            date = plugin_record[tlkey]['removal_date']
+            if isinstance(date, datetime.datetime):
+                date = date.date()
+            if isinstance(date, datetime.date):
+                date = date.isoformat()
+            plugin_record[tlkey]['removal_date'] = date
+    return plugin_record
```

Comment:
I don't really like that this function both returns plugin_record and mutates plugin_record.  But I'm not sure if there's an easy way to fix that.  It would be preferable as an API if it returned plugin_record but that would require copying plugin_record when it's not needed.  OTOH, mutating the plugin_record without returning it here won't fit in with the  way we're using it in load_collection_routing.